Automattic/simplenote-ios
# Add Notice Controller and Presenter to In App Notifications

### Fix
This PR Implements the controller class and Presenter class for in app notifications.  Implements #1178 

With the addition of the controller and presenter classes inserting Notice views is much easier, the notices respond to keyboard location, handle changing orientation of device, animate in, and animate to disappear

The notices will look much as they did in the Notice View PRs, as the appearance has not changed much.

<img src="https://cdn-std.droplr.net/files/acc_593859/oFV4pF" />

<img src="https://cdn-std.droplr.net/files/acc_593859/9H8NDQ" />

### Test
To test, first you will need to go to the bin and get the code to modify the the note option buttons to display notices.
You can find the presentation code [here](https://mc.a8c.com/pb/289a8/)

Once you have added that code to the SPNoteEditorViewController+extensions.swift file test out the following things:
QUEUEING
1. go to a note and tap on the information button to see a notice without an action
2. tap on the new note button to see a notice with an action, tap on the action to make sure that works (will print to the console)
3. Tap on the information button again, and while the notice is presenting tap on the new note button.  This should queue the notice w/action to present when the first notice is finished
4. Tap on the information button, after the notice appears tap on it again.  A duplicate notice should not get queued and then presented

KEYBOARD AND ORIENTATION
5. Tap on the information button again and while the notice is there, change the screen orientation, the notice should recenter it's self
6. Display a notice and then make the keyboard appear and disappear, then notice should track to the keyboard
7. Test on an iPad with floating keyboard.  If the keyboard is floating the notice will go to the bottom position (same as if the keyboard was hidden)

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

Note there is some weirdness with the commits, some of them are duplicated and some have been reverted because of an update to the package.resolved file that wouldn't go away.  That change was committed but has been reverted.

### Release
> These changes do not require release notes.
